### PR TITLE
UMPD3 checker: Check external repositories when they are modified by a branch.

### PR DESCRIPTION
**Pull request by AdamVoysey**
Currently, UM dependencies are checked by the UMDP3 checker on the trunk.

However, in a branch, the branch diff is used rather than the source files themselves. (This means the script only checks things that are changed by a branch, not the entire source tree.) 

This in turn means changes from the other dependencies are not checked until _after_ they are on trunk in a linked ticket. (This is because the branch diff contains only the changes to the sources, not the code changes themselves - which exist in a different repository.)

This pull request modifies the script to ensure dependencies are checked with the branch, when the script identifies that either the `rose-suite.conf` file, or the `HOST_SOURCE` environment variables have been modified. In that case, it inspects the full source of the external repository, as it would do on trunk.

### CODE REVIEW

**Review by james-bruten-mo — APPROVED** [review]
I've got nothing useful to add here, happy to approve

**Comment by AdamVoysey** [comment]
Can I give this PR a bump?

File: script_umdp3_checker/bin/umdp3_check.pl [hunk 1]
```diff
@@ -347,7 +347,85 @@
...
+                print $host_var_name
+                  . " modified in environment."
+                  . " Running full check on this repository\n";
+                push @extracts, $o_repo;
+            }
+        }
+
+    }
+
+    # enable trunkmode for sepecific repositories if rose-stem/rose-suite.conf
```
Comment (ericaneininger) [inline script_umdp3_checker/bin/umdp3_check.pl:390]:
```suggestion
    # enable trunkmode for specific repositories if rose-stem/rose-suite.conf
```

File: script_umdp3_checker/bin/umdp3_check.pl [hunk 2]
```diff
@@ -347,7 +347,85 @@
...
+# The @external_checks array contains the names of all the non-UM repositories
+# extracted by the UM which should also be checked.
+my @external_checks = ( "shumlib", "meta", "ukca" );
+my %filepath_mapping = ( 'meta' => 'um_meta' );
+my @extracts = ();
+
+if ( $trunkmode == 0 ) {
+    if ($suite_mode) {
+
+        # enable trunkmode for sepecific repositories if the environment does
```
Comment (ericaneininger) [inline script_umdp3_checker/bin/umdp3_check.pl:360]:
```suggestion
        # enable trunkmode for specific repositories if the environment does
```

File: script_umdp3_checker/bin/umdp3_check.pl [hunk 3]
```diff
@@ -347,7 +347,85 @@
...
+                  . " Running full check on this repository\n";
+                push @extracts, $o_repo;
+            }
+        }
+    }
+
+    # remove any duplicates
+    @extracts = keys { map { $_ => 1 } @extracts };
+
+    # If we capturede any changes, enable trunk-mode for those repositories.
```
Comment (ericaneininger) [inline script_umdp3_checker/bin/umdp3_check.pl:416]:
```suggestion
    # If we captured any changes, enable trunk-mode for those repositories.
```

**Review by ericaneininger — CHANGES_REQUESTED** [review]
A couple of typos in comments with suggested change.  Otherwise all good.  Thanks.

**Review by ericaneininger — APPROVED** [review]

**Pull request merged**